src/framework-ondata.js

```diff
@@ -312,13 +312,18 @@ function handleOutput( port, lastLm, state ) {
 
         // Got any edges out of this port? 
         if (port.listAttached().length > 0) {
+            var nodeInstance = port.nodeInstance;
             logger.debug('sending state', {
                 state: util.inspect(state,{depth:1}),
-                nodeInstance: port.nodeInstance
+                nodeInstance: nodeInstance
             });
 
             port.send( state );
             port.disconnect();
+
+            if (nodeInstance.isTransient) { 
+                 nodeInstance.deleteAllVnis();
```

Comment:
I think it would also be wise to modify deleteVni() in the vni-manager.js to delete the inputStates.  input-states.js does not seem to have a deleteAllInputStates method, and that module is a bit gnarly so it might be easiest to ask James to add that function, so that you can call it from deleteVni().